src/components/Swap.tsx

```diff
@@ -57,9 +58,29 @@ const Swap = () => {
   const toPurse = useAtomValue(toPurseAtom);
   const offers = useAtomValue(offersAtom);
   const [swapButtonStatus, setSwapButtonStatus] = useAtom(swapButtonStatusAtom);
+  const instanceIds = useAtomValue(instanceIdsAtom);
 
-  const assetsLoaded =
-    brandToInfo.size && metrics && governedParams && instanceId;
+  const anchorPetnames = [...instanceIds.keys()];
+  const areAnchorsLoaded =
+    anchorPetnames.length &&
+    anchorPetnames
+      .map(petname => {
+        const metrics = metricsIndex.get(petname);
+        const brand = metrics?.anchorPoolBalance.brand;
+        const brandInfo = brand && brandToInfo.get(brand);
+        const governedParams = governedParamsIndex.get(petname);
+
+        if (metrics && brandInfo && governedParams) {
+          return true;
+        }
+        return false;
+      })
+      .reduce(
+        (arePrevAssetsLoaded, isAssetLoaded) =>
+          arePrevAssetsLoaded && isAssetLoaded,
+        true
+      );
```

Comment:
I think this can all be simplified using `every`
```
const areAnchorsLoaded = anchorPetnames.every(petname => {
  const metrics = metricsIndex.get(petname);
  const brand = metrics?.anchorPoolBalance.brand;
  const brandInfo = brand && brandToInfo.get(brand);
  const governedParams = governedParamsIndex.get(petname);
  return metrics && brandInfo && governedParams;
}
```